### PR TITLE
#492 runnable_in_console option

### LINKED ISSUE #492: Add an option for restricting CommandSender
User may want to make command executable on Console, so adding `playerOnly: true/false`  option would be perfect for them.
HOWEVER, in this case, the user now have inner variable named `sender`  instead of `player`.

### PR DESCRIPTION
**Pull request by rkdrns4747**
[Please Check This Comment](https://github.com/TriggerReactor/TriggerReactor/issues/492#issuecomment-1176256789)

### CODE REVIEW

**Comment by rkdrns4747** [comment]
> https://github.com/TriggerReactor/TriggerReactor/commits/development_separate_listeners

This PR #493 may also have some effects from the `development_separate_listener` update.

**Comment by rkdrns4747** [comment]
This PR is scheduled to be closed due to rebasing to PR #497, as soon as PR #501 being resolved.

File: bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/bridge/AbstractBukkitWrapper.java [hunk 1]
```diff
@@ -28,6 +26,8 @@ public IEntity wrap(Entity entity) {
 
     public abstract IPlayer wrap(Player player);
 
+    public abstract IConsoleCommandSender wrap(ConsoleCommandSender sender);
```
Comment (wysohn) [inline bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/bridge/AbstractBukkitWrapper.java:29]: Is implementation for BukkitConsoleCommandSender differ between latest and legacy versions? Or we could just make it a concrete method

File: bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/AbstractJavaPlugin.java [hunk 2]
```diff
@@ -455,8 +458,9 @@ public IPlayer getPlayer(String string) {
             return null;
     }
 
-    public ICommandSender getConsoleSender() {
```
Comment (wysohn) [inline bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/AbstractJavaPlugin.java:458]: Same: leave it and change in next version

File: bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/AbstractJavaPlugin.java [hunk 3]
```diff
@@ -476,7 +480,7 @@ public HandlerList getHandlers() {
         }
     }
 
-    public Object createPlayerCommandEvent(ICommandSender sender, String label, String[] args) {
```
Comment (wysohn) [inline bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/AbstractJavaPlugin.java:479]: Same: leave it and change in next version

File: bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/BukkitTriggerReactorCore.java [hunk 4]
```diff
@@ -455,14 +456,19 @@ public IPlayer getPlayer(String string) {
...
+        return bukkit.getConsoleSender();
+    }
+
     @Override
     public Object createEmptyPlayerEvent(ICommandSender sender) {
         return bukkit.createEmptyPlayerEvent(sender);
     }
 
     @Override
-    public Object createPlayerCommandEvent(ICommandSender sender, String label, String[] args) {
```
Comment (wysohn) [inline bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/BukkitTriggerReactorCore.java:464]: Same: leave it and change in next version

**Review by wysohn — CHANGES_REQUESTED** [review]
Generally looking good

Perhaps it's better to push it to next version instead though

File: bukkit/latest/src/main/java/io/github/wysohn/triggerreactor/bukkit/bridge/BukkitWrapper.java [hunk 5]
```diff
@@ -11,4 +13,7 @@ public IPlayer wrap(Player player) {
         return new BukkitPlayer(player);
     }
 
+    @Override
+    public IConsoleCommandSender wrap(ConsoleCommandSender sender) {return new BukkitConsoleCommandSender(sender);}
```
Comment (nicosammito) [inline bukkit/latest/src/main/java/io/github/wysohn/triggerreactor/bukkit/bridge/BukkitWrapper.java:17]:
```suggestion
    public IConsoleCommandSender wrap(final ConsoleCommandSender sender) {return new BukkitConsoleCommandSender(sender);}
```

File: bukkit/legacy/src/main/java/io/github/wysohn/triggerreactor/bukkit/bridge/BukkitWrapper.java [hunk 6]
```diff
@@ -11,4 +13,6 @@ public IPlayer wrap(Player player) {
         return new BukkitPlayer(player);
     }
 
+    @Override
+    public IConsoleCommandSender wrap(ConsoleCommandSender sender) {return new BukkitConsoleCommandSender(sender);}
```
Comment (nicosammito) [inline bukkit/legacy/src/main/java/io/github/wysohn/triggerreactor/bukkit/bridge/BukkitWrapper.java:17]:
```suggestion
    public IConsoleCommandSender wrap(final ConsoleCommandSender sender) {return new BukkitConsoleCommandSender(sender);}
```

File: bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/bridge/BukkitConsoleCommandSender.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
+package io.github.wysohn.triggerreactor.bukkit.bridge;
+
+import io.github.wysohn.triggerreactor.core.bridge.IConsoleCommandSender;
+import org.bukkit.command.ConsoleCommandSender;
+
+public class BukkitConsoleCommandSender implements IConsoleCommandSender {
+    private final ConsoleCommandSender sender;
+    public BukkitConsoleCommandSender(ConsoleCommandSender sender){
```
Comment (nicosammito) [inline bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/bridge/BukkitConsoleCommandSender.java:8]:
```suggestion
    public BukkitConsoleCommandSender(final ConsoleCommandSender sender){
```

File: bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/bridge/BukkitConsoleCommandSender.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        super();
+        this.sender = sender;
+    }
+    @Override
+    public void sendMessage(String message) {
```
Comment (nicosammito) [inline bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/bridge/BukkitConsoleCommandSender.java:13]:
```suggestion
    public void sendMessage(final String message) {
```

File: bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/bridge/BukkitConsoleCommandSender.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+        sender.sendMessage(message);
+    }
+
+    @Override
+    public boolean hasPermission(String permission) {
```
Comment (nicosammito) [inline bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/bridge/BukkitConsoleCommandSender.java:18]:
```suggestion
    public boolean hasPermission(final String permission) {
```

File: bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/AbstractJavaPlugin.java [hunk 10]
```diff
@@ -476,7 +480,7 @@ public HandlerList getHandlers() {
...
+    public Object createCommandEvent(ICommandSender sender, String label, String[] args) {
```
Comment (nicosammito) [inline bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/AbstractJavaPlugin.java:483]:
```suggestion
    public Object createCommandEvent(final ICommandSender sender, final String label, final String[] args) {
```

File: bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/BukkitTriggerReactorCore.java [hunk 11]
```diff
@@ -455,14 +456,19 @@ public IPlayer getPlayer(String string) {
         return bukkit.getPlayer(string);
     }
 
+    @Override
+    public IConsoleCommandSender getConsoleCommandSender() {
+        return bukkit.getConsoleSender();
+    }
+
     @Override
     public Object createEmptyPlayerEvent(ICommandSender sender) {
```
Comment (nicosammito) [inline bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/BukkitTriggerReactorCore.java:465]:
```suggestion
    public Object createEmptyPlayerEvent(final ICommandSender sender) {
```

File: bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/BukkitTriggerReactorCore.java [hunk 12]
```diff
@@ -455,14 +456,19 @@ public IPlayer getPlayer(String string) {
...
         return bukkit.createEmptyPlayerEvent(sender);
     }
 
     @Override
-    public Object createPlayerCommandEvent(ICommandSender sender, String label, String[] args) {
-        return bukkit.createPlayerCommandEvent(sender, label, args);
+    public Object createCommandEvent(ICommandSender sender, String label, String[] args) {
```
Comment (nicosammito) [inline bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/main/BukkitTriggerReactorCore.java:470]:
```suggestion
    public Object createCommandEvent(final ICommandSender sender, final String label, final String[] args) {
```

File: bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/manager/trigger/CommandTriggerManager.java [hunk 13]
```diff
@@ -158,6 +169,28 @@ private void execute(Object context, Player player, String cmd, String[] args, C
...
+                return;
+            }
+        }
+
+        Map<String, Object> varMap = new HashMap<>();
+        varMap.put("isByPlayer", false);
+        varMap.put("sender", sender);
+        varMap.put("command", cmd);
+        varMap.put("args", args);
+        varMap.put("argslength", args.length);
```
Comment (nicosammito) [inline bukkit/src/main/java/io/github/wysohn/triggerreactor/bukkit/manager/trigger/CommandTriggerManager.java:189]:
```suggestion
        Map<String, Object> varMap = Map.of("isByPlayer", true, "sender", sender, "command", cmd, "args", args, "argslength", args.length);
```

File: core/src/main/java/io/github/wysohn/triggerreactor/core/manager/trigger/command/CommandTrigger.java [hunk 14]
```diff
@@ -53,6 +54,14 @@ public void setTabCompleters(ITabCompleter[] tabCompleters) {
         }
     }
 
+    public boolean isConsoleAvailable(){
+        return console_available;
+    }
+
+    public void setConsoleAvailable(boolean bool){
```
Comment (nicosammito) [inline core/src/main/java/io/github/wysohn/triggerreactor/core/manager/trigger/command/CommandTrigger.java:61]:
```suggestion
    public void setConsoleAvailable(final boolean bool){
```

**Comment by wysohn** [comment]
this branch is stale for long time and is unlikely cleanly merged

Suggested to create a new branch and replicate the changes there

**Pull request closed without merging**